lib/unification/Syntactic.ml

```diff
@@ -212,29 +212,40 @@ and quasi_solved env stack x non_arrow s =
 
 (* Non proper equations
    'x ≡ 'y
+   To include a non propre equations, we need to be sure that the dependency created between variable is a DAG. Therefore, we use the Variable.compare, to order the dependency.
 *)
 and non_proper env stack (x : Variable.t) non_arrow_x (y : Variable.t) non_arrow_y =
   Trace.with_span ~__FUNCTION__ ~__LINE__ ~__FILE__ __FUNCTION__ (fun _sp ->
   match
     (Env.representative ~non_arrow:non_arrow_x env x,
      Env.representative ~non_arrow:non_arrow_y env y) with
-  | (V x' | NAR x'), (V y' | NAR y') when Variable.equal x' y' -> process_stack env stack
-  | NAR y', (V x' | NAR x') | V x', NAR y' ->
+  | (V x' | NAR x' | E (x', _)), (V y' | NAR y' | E (y', _)) when Variable.equal x' y' -> process_stack env stack
```

Comment:
Is the case with `V x'`, `NAR y'` with `Variable.equal x' y'` possible ? Should we assert it isn't ?